src/ds/d400/d400-options.h

```diff
@@ -100,52 +109,101 @@ namespace librealsense
         hw_monitor& _hwm;
         sensor_base* _sensor;
         std::weak_ptr<limits_option> _gain_limit_toggle;
+        bool _ae_gain_limits_new_opcode;
     };
 
     // Auto-Limits Enable/ Disable
     class limits_option : public option
     {
     public:
 
-        limits_option(rs2_option option, option_range range, const char* description, hw_monitor& hwm) :
-            _option(option), _toggle_range(range), _description(description), _hwm(hwm) {};
+        limits_option( rs2_option option,
+                       option_range range,
+                       const char * description,
+                       hw_monitor & hwm,
+                       bool ae_gain_limits_new_opcode = false )
+            : _option( option )
+            , _toggle_range( range )
+            , _description( description )
+            , _hwm( hwm )
+            , _ae_gain_limits_new_opcode( ae_gain_limits_new_opcode ){};
 
         virtual void set(float value) override
         {
             auto set_limit = _cached_limit;
             if (value == 0) // 0: gain auto-limit is disabled, 1 : gain auto-limit is enabled (all range 16-248 is valid)
                 set_limit = 0;
 
-            command cmd_get(ds::AUTO_CALIB);
-            cmd_get.param1 = 5;
-            std::vector<uint8_t> ret = _hwm.send(cmd_get);
-            if (ret.empty())
-                throw invalid_value_exception("auto_exposure_limit_option::query result is empty!");
-
-            command cmd(ds::AUTO_CALIB);
-            cmd.param1 = 4;
-            cmd.param2 = *(reinterpret_cast<uint32_t*>(ret.data()));
-            cmd.param3 = static_cast<int>(set_limit);
-            if (_option == RS2_OPTION_AUTO_EXPOSURE_LIMIT_TOGGLE)
+            if( _ae_gain_limits_new_opcode )
```

Comment:
2 features?

Reply:
I think we need 2 features (see another comment) but here it is splitting to 2 options.
At least using helper functions.
```
if( _new_opcode )
    set_using_new_opcode()
else
    set_using_old_opcode()
```